### PR TITLE
New article: How to integrate 3rd-party payment gateway

### PR DESCRIPTION
**Pull request by rap1ds**
New article: How to integrate 3rd-party payment gateway

**Review:** This article (structure, language, scope etc) has been already heavily reviewed (by Juho, Thomas, Mira), so I'm not really expecting a lot of comments on that front. However, I'd appreciate review on:

* Correct markdown formatting
* Review of the sequence diagrams (which I assume Juho, Thomas, Mira didn't review that well)
* Review from a technical perspective. Is everything written technically correct?

### CODE REVIEW

File: src/docs/background/how-to-integrate-3rd-party-payment-gateway/index.md [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+Sharetribe Flex provides out-of-the-box integration with Stripe. To reach markets not supported by Stripe, relying on another payment gateway can be crucial. This guide describes on a high-level, without going into the details of specific payment gateways, how to integrate any 3rd-party payment gateway (such as [PayPal Commerce Platform](https://www.paypal.com/us/business/platforms-and-marketplaces), [MANGOPAY](https://www.mangopay.com/marketplaces/), or [Adyen for Platforms](https://www.adyen.com/our-solution/marketplaces-and-platforms)) with Sharetribe Flex.
+
+## Prerequisites
+
+Before reading this guide, you should be familiar with the following Sharetribe Flex features:
+
+* [Privileged transitions](https://www.sharetribe.com/docs/background/privileged-transitions/)
+* [Events](https://www.sharetribe.com/docs/references/events/)
+* [Reacting to events](https://www.sharetribe.com/docs/cookbook-events/reacting-to-events/)
+* [Extended data](https://www.sharetribe.com/docs/references/extended-data/)
```
Comment (kpuputti) [inline src/docs/background/how-to-integrate-3rd-party-payment-gateway/index.md:20]: If you remove the `https://www.sharetribe.com/docs ` prefix from the URLs, they also work locally and in deploy previews.

File: src/docs/background/how-to-integrate-3rd-party-payment-gateway/index.md [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+
+## Marketplace payment flow
+
+In this section we illustrate a marketplace payment flow in high-level and briefly discuss each stage. Later in this article, we'll go through how to integrate each state with your Flex-powered marketplace.
+
+In a nutshell, a payment flow in a marketplace contains five significant steps. The following diagram illustrates a timeline of these steps:
+
+![alt_text](payment-flow.png "Marketplace payment flow")
```
Comment (kpuputti) [inline src/docs/background/how-to-integrate-3rd-party-payment-gateway/index.md:28]: Alt texts are important, especially in documentation!

**Review by kpuputti — COMMENTED** [review]
I'm still reading the article, but these comments might be useful already.

File: src/docs/background/how-to-integrate-3rd-party-payment-gateway/index.md (hunk 2 above)
Comment (rap1ds) [reply src/docs/background/how-to-integrate-3rd-party-payment-gateway/index.md:28]: Yes for sure! I'll fix this. I didn't mean to leave alt texts like this, but in fact, I wasn't familiar with the syntax and thought that the "Marketplace payment flow" would be the alt text, but it seems that that becomes the title of the image, not alt text.

File: src/docs/background/how-to-integrate-3rd-party-payment-gateway/index.md [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+### Using privileged transitions
+
+With privileged transitions, you need to make a new endpoint to your backend server. Your marketplace front-end should call this new backend endpoint and not the Flex API directly.
+
+If you use Flex Template for Web, you can add a new endpoint by adding it to the [API router][ftw-daily-api-router].
+
+The new server endpoint should call the payment gateway API to do the payment action and the Flex Marketplace API to transition the transaction.
+
+The following diagram shows the call sequence between the marketplace and the APIs. First, a user makes a transaction request using the marketplace front end, after which the backend calls the Flex Marketplace API to initiate the transaction.Next, the marketplace backend calls the payment gateway’s API, and finally the Flex Marketplace API is called again after the payment action is completed.
+
```
Comment (kpuputti) [inline src/docs/background/how-to-integrate-3rd-party-payment-gateway/index.md:230]: Missing space before "Next"

> initiate the transaction.Next, the

**Pull request merged**